docs/content/InputField.mdx

```diff
@@ -0,0 +1,192 @@
+---
+title: InputField
+status: Alpha
+source: https://github.com/primer/react/blob/main/src/InputField.tsx
+storybook: '/react/storybook?path=/story/forms-inputfield--text-input-field'
+---
+
+import {InputField, TextInputWithTokens} from '@primer/react'
+import {ComponentChecklist} from '../src/component-checklist'
+
+The `InputField` component is used to render a labelled text input and, optionally, associated validation text and hint text.
+
+## Anatomy
+
+### InputField
+
+The container that handles the layout and passes the relevant IDs and ARIA attributes it's children.
+
+### InputField.Label
+
+A `InputField.Label` must be passed, but it may be visually hidden.
+
+### InputField.Caption
+
+If this field needs additional context, a `InputField.Caption` may be used to render hint text.
```

Comment:
Shall we switch to active for these descriptions? E.g. `InputField.Caption` can be used to render hint text for fields that require additional context`